csp_billing_adapter/adapter.py

```diff
@@ -214,7 +216,23 @@ def main() -> None:
             log
         )
 
-        initial_adapter_setup(pm.hook, config, log)
+        cache, csp_config = initial_adapter_setup(pm.hook, config, log)
+
+        try:
+            # Test metering API access
+            process_metering(
+                config,
+                cache,
+                pm.hook,
+                empty_metering=True,
+                dry_run=True
+            )
+        except Exception as error:
+            raise CSPBillingAdapterException(
+                f'Fatal error while validating metering API access: {error}'
+            )
+
+        time.sleep(config.query_interval)  # wait 1 cycle for usage data
```

Comment:
So we are making the assumption that the application will be up, running, and reporting within the configured interval? Trying to confirm my understanding.

Reply:
Yes, the expectation is that the application will be up and provide a usage record within the first cycle. If this is not the case the adapter still continues to run but an error will be added to the error list in "API".

This solves the current problem where an error shows up at startup due to no usage records.